src/components/modal/index.stories.mdx

```diff
@@ -0,0 +1,150 @@
+import { Canvas, Meta, Story } from '@storybook/addon-docs';
+import { action } from '@storybook/addon-actions';
+import { useDisclosure } from '@chakra-ui/react';
+
+import Button from '../button';
+import Box from '../box';
+
+import Modal from './index';
+
+<Meta title="Components/Data display/Modal" component={Modal} />
+
+export const ModalTemplate = (args) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  return (
+    <Box>
+      <Button onClick={onOpen}>Open</Button>
+      <Modal isOpen={isOpen} onClose={onClose} {...args}>
+        <Box p="4">
+          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer
+          elementum ultricies vestibulum. Sed odio diam, tristique id velit sit
+          amet, sodales aliquet magna. Fusce vitae orci vel nulla vehicula
+          condimentum. Maecenas sollicitudin volutpat tellus, laoreet fringilla
+          nibh. Vivamus luctus enim id quam feugiat, maximus porttitor justo
+          ultrices. Nam tellus erat, porta nec consectetur eget, aliquet id
+          purus. Suspendisse at elementum arcu. Suspendisse ac lobortis velit.
+          Phasellus malesuada egestas est sit amet venenatis.
+        </Box>
+      </Modal>
+    </Box>
+  );
+};
+
+# Modal
+
+<Canvas>
+  <Story
+    name="Modal"
+    args={{
+      isCentered: true,
+      size: 'md',
+      motionPreset: 'scale',
```

Comment:
it's fine to expose but I would not make it visible in the docs. Designers will have crazy ideas otherwise. I'm also ok to not expose it at all